### PR TITLE
MAT-6313 Feature flag for Execution of test cases containing observat…

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
…ions or stratifications

## MADiE PR

Jira Ticket: [MAT-0000](https://jira.cms.gov/browse/MAT-0000)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

**Comment by nmorasb** [comment]
Can you please link the JIRA ticket

File: src/components/editTestCase/qiCore/EditTestCase.tsx [hunk 1]
```diff
@@ -271,6 +273,26 @@ const EditTestCase = (props: EditTestCaseProps) => {
     measure?.measureSet?.acls,
     measure?.measureMetaData?.draft
   );
+  const [hasObservationOrStratification, setHasObservationOrStratification] =
+    useState(false);
+  useEffect(() => {
```
Comment (nmorasb) [inline src/components/editTestCase/qiCore/EditTestCase.tsx]: I don't think it was mentioned in the ticket, aside from being in the QDM Test Cases epic..but I think we only want to block execution of strats or obs for QDM, not QI-Core. Please confirm with Christina and update if needed. (Also we can update the ticket)

Comment (sb-cecilialiu) [reply src/components/editTestCase/qiCore/EditTestCase.tsx]: @nmorasb QI-Core buttons are reverted, thanks a lot Nate!

File: src/components/testCaseLanding/common/runTestsButton/RunTestsButton.tsx [hunk 2]
```diff
@@ -22,6 +34,26 @@ export default function RunTestButton(props: RunTestButtonProps) {
...
+      measure?.model === Model.QDM_5_6 &&
+      featureFlags?.disableRunTestCaseWithObservStrat
+    ) {
+      const groups: Group[] = measure?.groups;
+      groups?.forEach((group) => {
+        const measureObservations: MeasureObservation[] =
+          group?.measureObservations;
+        const measureStratifications: Stratification[] = group?.stratifications;
+        if (
+          measureObservations ||
```
Comment (nmorasb) [inline src/components/testCaseLanding/common/runTestsButton/RunTestsButton.tsx]: Can we also do the length check on observations as well? Not sure if they would add observations and then remove them if we might end up with an empty observations array which would then disable the button

File: src/components/editTestCase/qdm/EditTestCase.tsx [hunk 3]
```diff
@@ -90,6 +97,26 @@ const EditTestCase = () => {
...
+    useState(false);
+  useEffect(() => {
+    if (featureFlags?.disableRunTestCaseWithObservStrat) {
+      const groups: Group[] = measure?.groups;
+      groups?.forEach((group) => {
+        const measureObservations: MeasureObservation[] =
+          group?.measureObservations;
+        const measureStratifications: Stratification[] = group?.stratifications;
+        if (
+          measureObservations ||
```
Comment (nmorasb) [inline src/components/editTestCase/qdm/EditTestCase.tsx]: same thing here - let's check the length of observations array

File: src/components/testCaseLanding/common/runTestsButton/RunTestsButton.tsx (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/testCaseLanding/common/runTestsButton/RunTestsButton.tsx]: @nmorasb done.

File: src/components/editTestCase/qdm/EditTestCase.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/editTestCase/qdm/EditTestCase.tsx]: @nmorasb done, thanks!

**Review by nmorasb — APPROVED** [review]

**Pull request merged**